sweep/sweeper.go

```diff
@@ -979,6 +1021,100 @@ func (s *UtxoSweeper) handlePendingSweepsReq(
 	return pendingInputs
 }
 
+// BumpFee allows bumping the fee of an input being swept by the UtxoSweeper
+// according to the provided fee preference. The new fee preference will be used
+// for a new sweep transaction of the input that will act as a replacement
+// transaction (RBF) of the original sweeping transaction, if any.
+//
+// NOTE: This currently doesn't do any fee rate validation to ensure that a bump
+// is actually successful. The responsibility of doing so should be handled by
+// the caller.
+func (s *UtxoSweeper) BumpFee(input wire.OutPoint,
+	feePreference FeePreference) (chan Result, error) {
+
+	// Ensure the client provided a sane fee preference.
+	if _, err := s.feeRateForPreference(feePreference); err != nil {
+		return nil, err
+	}
+
+	responseChan := make(chan *bumpFeeResp, 1)
+	select {
+	case s.bumpFeeReqs <- &bumpFeeReq{
+		input:         input,
+		feePreference: feePreference,
+		responseChan:  responseChan,
+	}:
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+
+	select {
+	case response := <-responseChan:
+		return response.resultChan, response.err
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+}
+
+// handleBumpFee handles an incoming request to bump the fee rate of a given
+// input.
+func (s *UtxoSweeper) handleBumpFeeReq(req *bumpFeeReq,
+	bestHeight int32) (chan Result, error) {
+
+	// If the UtxoSweeper is already trying to sweep this input, then we can
+	// simply just increase its fee rate. This will allow the input to be
+	// batched with others which also have a similar fee rate, creating a
+	// higher fee rate transaction that replaces the original input's
+	// sweeping transaction.
+	pendingInput, ok := s.pendingInputs[req.input]
+	if !ok {
+		return nil, lnwallet.ErrNotMine
+	}
+
+	return s.handleRBFBump(req, pendingInput, bestHeight)
+}
+
+// handleRBFBump handles a bump fee request by simply updating the inputs fee
+// preference. Currently, no validation is done on the new fee preference to
+// ensure it will properly create a replacement transaction.
+//
+// TODO(wilmer):
+//   * Validate fee preference to ensure we'll create a valid replacement
+//     transaction to allow the new fee rate to propagate throughout the
+//     network.
+//   * Ensure we don't combine this input with any other unconfirmed inputs that
+//     did not exist in the original sweep transaction, resulting in an invalid
+//     replacement transaction.
+func (s *UtxoSweeper) handleRBFBump(req *bumpFeeReq, input *pendingInput,
+	bestHeight int32) (chan Result, error) {
```

Comment:
Why return a `chan Result`? It is not used other than in the test it seems and maybe there you can use the original input registration result channel? (yagni?)

Reply:
I was thinking of the case where a new caller wants to bump the fee of the input, but they weren't the one who submitted the input request, so they wouldn't be notified of when the input gets swept.

Reply:
Ok, I can see the use case. I think I would have added it only when actually needed.